R/autoplot.pop_data.R

```diff
@@ -110,15 +113,33 @@ density_plot <- function(
       ggplot2::labs(fill = strata)
   }
   if (log) {
+
+    min_nonzero_val =
+      object %>%
+      filter(value > 0) %>%
+      pull(value) %>%
+      min()
+
+    max_val =
+      object %>%
+      pull(value) %>%
+      max()
+
+   breaks1 = c(0, 10^seq(
+      min_nonzero_val %>% log10() %>% floor(),
+      max_val %>% log10() %>% ceiling()))
+
     plot1 <- plot1 +
-      #ggplot2::scale_x_log10() +
-      ggplot2::scale_y_continuous(trans = scales::pseudo_log_trans(sigma = 0.01),
-                                  breaks = c(-1, -0.1, 0, 0.1, 1, 10), minor_breaks = NULL) +
-      ggplot2::labs(
-        title = "Distribution of Cross-sectional Antibody Responses (Log transformed)",
-        x = "Log10(Antibody Response Value)",
-        y = "Frequency"
-      )
+      ggplot2::scale_x_continuous(
+        labels = scales::label_comma(),
+        transform = scales::pseudo_log_trans(
+          sigma = min_nonzero_val/10,
+          base = 10),
+        breaks = breaks1
+      ) +
+      ggplot2::labs(title = "Distribution of Cross-sectional Antibody Responses (Log transformed)",
+                    x = "Log10(Antibody Response Value)",
+                    y = "Frequency")
```

Comment:
I made some changes to fine-tune the `sigma` parameter so that the full range of nonzero values gets treated as non-zero. I also restored the minor breaks and made sure the x-axis ticks go all the way through the data range:

![image](https://github.com/UCD-SERG/serocalculator/assets/2474437/c485db55-17e9-4ea4-9d2b-19ce6cbe76aa)

after:

![image](https://github.com/UCD-SERG/serocalculator/assets/2474437/116ef150-92e1-44e3-bbc5-420d1debc943)

Reply:
This is in order.